code/game/machinery/biogenerator.dm

```diff
@@ -329,6 +329,18 @@
 	speed_coefficient = 2/manipcount
 	biomass_coefficient = 3*lasercount
 
+/obj/machinery/biogenerator/AltClick(mob/user)
+	if(!user.incapacitated() && Adjacent(user) && user.dexterity_check())
+		eject_produce()
+		return
+	return ..()
+
+/obj/machinery/biogenerator/CtrlClick(mob/user)
+	if(!user.incapacitated() && Adjacent(user) && beaker && user.dexterity_check())
```

Comment:
That is a really good point

Honestly I feel like instead of making all these checks over and over we should at some point have some kind of high-level proc for 'user can use a machine' that makes the big three checks, then when machinery inherits it, add broken and power checks, etc.